transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/OpeningHandshakeIT.java

```diff
@@ -0,0 +1,320 @@
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+/**
+ * RFC-6455, section 4.1 "Client-Side Requirements"
+ * RFC-6455, section 4.2 "Server-Side Requirements"
+ */
+public class OpeningHandshakeIT {
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/opening");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/path"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                        .done()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/preflight"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(gateway).around(timeout);
+
+    // TODO:
+    // proxy => HTTP CONNECT w/ optional authorization, auto-configuration via ws://, wss://
+    // TLS (not SSL) w/ SNI for wss://
+
+    @Test
+    @Specification({
+        "connection.established/handshake.request"
+        })
+    public void shouldEstablishConnection() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("This spec test will not work. Preflight cookie request not 'a thing'. Talk to engineers.")
+    @Specification({
+        "request.header.cookie/handshake.request"
+        })
+    public void shouldEstablishConnectionWithCookieRequestHeader() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "request.headers.random.case/handshake.request"
+        })
+    public void shouldEstablishConnectionWithRandomCaseRequestHeaders() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "response.headers.random.case/handshake.request"
+        })
+    public void shouldEstablishConnectionWithRandomCaseResponseHeaders() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "request.header.origin/handshake.request"
+        })
+    public void shouldEstablishConnectionWithRequestHeaderOrigin() throws Exception {
```

Comment:
There should be a negative test for the case where the origin header value is not accepted by the server (which according to RFS 6455 section 10.2 should result in HTTP 403 Forbidden status code). This test needs to be added first to the wsn spec project.